lib/avo.rb

```diff
@@ -39,7 +39,7 @@ def initialize(resource_name)
     private
 
     def missing_resource_message(resource_name)
-      name = resource_name.to_s.downcase
+      name = resource_name.to_s.gsub(/([a-z])([A-Z])/, '\1_\2').downcase
```

Comment:
I think regex is overkill for this use case, WDYT about using the [`underscore`](https://apidock.com/rails/String/underscore) method?

```suggestion
      name = resource_name.to_s.underscore
```